### PR TITLE
Rename Result to GravatarResult

### LINKED ISSUE #296: Drop Result type in favor of kotlin.Result
Gravata.Result type doesn't add anything extra and as per this [post](https://kotlinlang.org/docs/api-guidelines-simplicity.html#reuse-existing-concepts) this will reduce the mental complexity and shrink the size of our API.

### PR DESCRIPTION
**Pull request by AdamGrzybkowski**
Closes #296 

### Description

As per the discussion in #296 - the `Result` could be easily confused with `kotlin.Result` so we won't to rename it for less confusion and better autocompletion. 

### Testing Steps

CI should be fine. Check the docs.

### CODE REVIEW

File: gravatar/src/main/java/com/gravatar/services/GravatarResult.kt [hunk 1]
```diff
@@ -3,15 +3,15 @@ package com.gravatar.services
...
+public sealed class GravatarResult<T, E> {
     /**
      * Represents a successful fetch operation.
      *
      * @param T The type of the value fetched
      * @param E The error type
      * @property value The fetched value
      */
-    public data class Success<T, E>(public val value: T) : Result<T, E>()
+    public data class Success<T, E>(public val value: T) : GravatarResult<T, E>()
```
Comment (AdamGrzybkowski) [inline gravatar/src/main/java/com/gravatar/services/GravatarResult.kt:14]: Should we drop the `data` here?

File: gravatar/src/main/java/com/gravatar/services/AvatarService.kt [hunk 2]
```diff
@@ -51,15 +51,17 @@ public class AvatarService(private val okHttpClient: OkHttpClient? = null) {
 
     /**
      * Uploads an image to be used as Gravatar avatar.
-     * This method will catch any exception that occurs during the execution and return it as a [Result.Failure].
+     * This method will catch any exception that occurs during
+     * the execution and return it as a [Result.Failure].
```
Comment (hamorillo) [inline gravatar/src/main/java/com/gravatar/services/AvatarService.kt]:
```suggestion
     * the execution and return it as a [GravatarResult.Failure].
```

File: gravatar/src/main/java/com/gravatar/services/GravatarResult.kt (hunk 1 above)
Comment (hamorillo) [reply gravatar/src/main/java/com/gravatar/services/GravatarResult.kt:14]: In this case, I think we can keep it, or do you see a risk in that?

**Review by hamorillo — APPROVED** [review]
🚀

File: gravatar/src/main/java/com/gravatar/services/GravatarResult.kt (hunk 1 above)
Comment (AdamGrzybkowski) [reply gravatar/src/main/java/com/gravatar/services/GravatarResult.kt:14]: I don't think we would add more properties to it, so I do also think it's safe.

**Pull request merged**